framework/decode/file_processor.cpp

```diff
@@ -413,7 +416,8 @@ bool FileProcessor::ProcessBlocks()
                     // end of file warning when the file is at EOF without an error. For this case (the normal EOF case)
                     // we print nothing at EOF, or print an error message and set the error code directly when not at
                     // EOF.
-                    GFXRECON_LOG_ERROR("Failed to read block header");
+                    GFXRECON_LOG_ERROR(
+                        "Failed to read block header (frame %lu block %lu)", current_frame_number_, block_index_);
```

Comment:
Dangerous 64/32 bit format codes? There is an explicit `uint32_t` and an explicit `uint64_t`, both given the loosy-goosey format code `%lu` (`unsigned long`) which potentially means different things on different compilers/platforms/32/64 bit build versions. At least on some build on some platform we care about, one of `uint32_t` or `uint64_t` is not going to be a `%lu`.

Here is a belt and braces version:

```cpp
GFXRECON_LOG_ERROR("Failed to read block header (frame %u block %llu)", static_cast<unsigned>(current_frame_number_), static_cast<unsigned long long>(block_index_));
```
I don't think you need the casts on 32 bit or 64 bit platforms we care about but they won't hurt.

Reply:
Frame count is uint32, so I switched to `%u` which should be universal.  Block index was uint64_t so I switched to `%" PRIu64"` which also is universal without having to do any casts.

Reply:
Oh cool, I didn't actually know about those format specifiers. They do exactly what we want (https://www.phind.com/search?cache=llnuuji1691dcyfheuxw3lvs, https://en.cppreference.com/w/cpp/header/cinttypes).

As long as nobody changes the types of these members later, the casts will never be needed.